pallets/transaction-multi-payment/src/lib.rs

```diff
@@ -115,164 +148,136 @@ decl_error! {
 			}
 			Err(Error::<T>::AlreadyAccepted.into())
 		}
 
 		/// Remove currency from the list of supported currencies
 		/// Only selected members can perform this action
-		#[weight = (<T as Config>::WeightInfo::remove_currency(), Pays::No)]
-		pub fn remove_currency(origin, currency: AssetId) -> DispatchResult{
-
+		#[pallet::weight((<T as Config>::WeightInfo::remove_currency(), DispatchClass::Normal, Pays::No))]
+		pub fn remove_currency(origin: OriginFor<T>, currency: AssetId) -> DispatchResultWithPostInfo {
 			let who = ensure_signed(origin)?;
 
-			ensure!(
-				currency != CORE_ASSET_ID,
-				Error::<T>::CoreAssetNotAllowed
-			);
+			ensure!(currency != CORE_ASSET_ID, Error::<T>::CoreAssetNotAllowed);
 
 			// Only selected accounts can perform this action
-			ensure!(
-				Self::authorities().contains(&who),
-				Error::<T>::NotAllowed
-			);
-
-			if AcceptedCurrencies::mutate(|x| x.remove(&currency)) {
-				Self::deposit_event(RawEvent::CurrencyRemoved(who, currency));
-				return Ok(());
+			ensure!(Self::authorities().contains(&who), Error::<T>::NotAllowed);
+
+			if AcceptedCurrencies::<T>::mutate(|x| x.remove(&currency)) {
+				Self::deposit_event(Event::CurrencyRemoved(who, currency));
+				return Ok(().into());
 			}
 
 			Err(Error::<T>::UnsupportedCurrency.into())
 		}
 
 		/// Add an account as member to list of authorities who can manage list of accepted currencies
-		#[weight = (<T as Config>::WeightInfo::add_member(), Pays::No)]
-		pub fn add_member(origin, member: T::AccountId) -> DispatchResult{
+		#[pallet::weight((<T as Config>::WeightInfo::add_member(), DispatchClass::Normal, Pays::No))]
+		pub fn add_member(origin: OriginFor<T>, member: T::AccountId) -> DispatchResultWithPostInfo {
 			ensure_root(origin)?;
 
-			ensure!(
-				! Self::authorities().contains(&member),
-				Error::<T>::AlreadyMember
-			);
+			ensure!(!Self::authorities().contains(&member), Error::<T>::AlreadyMember);
 
 			Self::add_new_member(&member);
 
-			Self::deposit_event(RawEvent::MemberAdded(member));
+			Self::deposit_event(Event::MemberAdded(member));
 
-			Ok(())
+			Ok(().into())
 		}
 
-		/// Add an account as member to list of authorities who can manage list of accepted currencies
-		#[weight = (<T as Config>::WeightInfo::remove_member(), Pays::No)]
-		pub fn remove_member(origin, member: T::AccountId) -> DispatchResult{
+		#[pallet::weight((<T as Config>::WeightInfo::remove_member(), DispatchClass::Normal, Pays::No))]
+		pub fn remove_member(origin: OriginFor<T>, member: T::AccountId) -> DispatchResultWithPostInfo {
 			ensure_root(origin)?;
 
-			ensure!(
-				Self::authorities().contains(&member),
-				Error::<T>::NotAMember
-			);
+			ensure!(Self::authorities().contains(&member), Error::<T>::NotAMember);
 
 			Authorities::<T>::mutate(|x| x.retain(|val| *val != member));
 
-			Self::deposit_event(RawEvent::MemberRemoved(member));
-
-			Ok(())
-		}
-
-		fn on_runtime_upgrade() -> frame_support::weights::Weight {
```

Comment:
Is it the case that we should remove older upgrades?

Reply:
yes, as i understood - once you upgrade - the migration should be removed.  That's why i commented out the migration code in this pallet.